File: scripts/refresh_countries.mjs
import { writeFileSync } from 'fs';
import stringify from 'json-stable-stringify';

writeFileSync(
  'query/assets/countries.json',
  stringify(
    await (
      await fetch(
        'https://static.openfoodfacts.org/data/taxonomies/countries.json',
      )
    ).json(),
    { space: 2 },
  ),
);

File: scripts/refresh_countries.py
"""Refreshes the countries.json file from the currently published Open Food Facts taxonomy"""

import json
import requests


res = requests.get("https://static.openfoodfacts.org/data/taxonomies/countries.json")
response = res.json()
with open("query/assets/countries.json", "w", encoding="utf-8") as f:
    json.dump(response, f, sort_keys=True, ensure_ascii=False, indent=2)
